src/services/base.service.js

```diff
@@ -51,6 +108,73 @@ const baseSearchById = async (args, resourceType) => {
   return new dataType(result);
 };
 
+/**
+ * Searches using query parameters.
+ * @param {*} args The args from the request.
+ * @param {*} req The Express request object. This is used by the query builder.
+ * @param {*} resourceType The resource type we are searching on.
+ * @param {*} paramDefs Optional parameter definitions for the specific resource types. Specific
+ *                      resource services should call this and pass
+ * @returns Search set result bundle
+ */
+const baseSearch = async (args, { req }, resourceType, paramDefs = {}) => {
+  // grab the schemas for the data type and Bundle to use for response
+  const dataType = resolveSchema(args.base_version, resourceType.toLowerCase());
+  const Bundle = resolveSchema(args.base_version, 'bundle');
+
+  // wipe out params since the 'base_version' here breaks the query building
+  req.params = {};
+
+  // build result bundle. default to an empty result
+  const searchBundle = new Bundle({
+    type: 'searchset',
+    meta: { lastUpdated: new Date().toISOString() },
+    total: 0
```

Comment:
NIT: I couldn't find anything in the FHIR spec for whether or not this matters, but maybe it's better consistency to have a default of `entry: []` instead of omitting is so that the empty bundles just have an empty entry array rather than a non-existent one

Reply:
Resolved in slack. HAPI doesn't return an empty entry array, so we wont either.